packages/@react-aria/overlays/src/useOverlayPosition.ts

```diff
@@ -184,9 +184,11 @@ export function useOverlayPosition(props: AriaPositionProps): PositionAria {
     };
 
     visualViewport?.addEventListener('resize', onResize);
+    visualViewport?.addEventListener('scroll', onResize);
```

Comment:
This is because mobile safari scrolls the visual viewport when focusing an input, so the input is centered. This happens after the resize event, and thus the available space for the overlay can change so we need to reposition in this case as well.